plugins/inputs/execd/README.md

```diff
@@ -25,12 +29,50 @@ STDERR from the process will be relayed to Telegraf as errors in the logs.
   ## Valid values are:
   ##   "none"    : Do not signal anything. (Recommended for service inputs)
   ##               The process must output metrics by itself.
-  ##   "STDIN"   : Send a newline on STDIN. (Recommended for gather inputs)
+  ##   "STDIN"   : Send a newline on STDIN. (Recommended for gather inputs).
+  ##               Note that it is standard input from the perspective of the
+  #                daemon, not telegraf
   ##   "SIGHUP"  : Send a HUP signal. Not available on Windows. (not recommended)
   ##   "SIGUSR1" : Send a USR1 signal. Not available on Windows.
   ##   "SIGUSR2" : Send a USR2 signal. Not available on Windows.
   signal = "none"
 
+  ## Startup string
+  ##
+  ## The standard way of configuring a daemon is through the command line.
+  ## If the daemon requires additional configuratin, this is normally done
+  ## through its own configuration file.  In some cases, the command line
+  ## might provide the daemon with a path to such a configuration.
+  ##
+  ## As an alternative, some daemons may allow configuration through their
+  ## standard input, as if a user had typed them or redirected them from
+  ## a file.  Exed allows this by setting a write_on_start parameter.
+  ## 
+  ## !IMPORTANT! This setting is ignored unless signal="STDIN".
+  ## !IMPORTANT! Newlines are not automatic - add \n if you want one.  
+  ##
+  ## write_on_start is sent ONCE when the daemon is started.  If this string is
+  ## not specified or is zero-length, nothing is sent.  Note that TOM supports
+  ## multi-line string inputs.
+  write_on_start=""
```

Comment:
Yeah.  I didn't come up with a great solution for that.  If it's something like a text mode input, maybe it's \n\n but if it's something like xml or json then that's a problem.  For signaling methods other than STDIN, actually closing STDIN would work in some circumstances but probably not others.  In the end, I figured that how this works is really a contract between the daemon and the config.

One of my goals for the startup string is for services like BACnet where on startup you want to send a bunch of SUBSCRIBE-like commands, then just listen for metrics to stream back asynchronously.

Reply:
Hmm. It'd be great if it was out of band or something.. like if the execd program ignored it and didn't deal with the config it'd still work well. Something like base64 encoding it and passing it all as a command line argument (though that might have size limits)..

Reply:
I don't want to base64 encode it or anything like that because part of the idea here is to be able to re-use existing daemons/applications with little or no modification. That's why I went for the simplicity of it's just sending text to the process's stdin just as if you were typing it by hand.  If you still disagree, let's discuss ....